MailCore/Cache/Attachments/AttachmentsManagerWorker/AttachmentsManagerWorker.swift

```diff
@@ -377,6 +370,44 @@ extension AttachmentsManagerWorker: AttachmentsManagerWorkable {
         await updateDelegate?.contentWillChange()
     }
 
+    public func processTextAttachment(_ attachment: TextAttachable) async {
+        // process attachment
+        let textAttachment = await attachment.textAttachment
+
+        // mutate Draft
+        await backgroundRealm.execute { realm in
+            try? realm.write {
+                guard let draftInContext = realm.object(ofType: Draft.self, forPrimaryKey: self.draftLocalUUID) else {
+                    return
+                }
+
+                // Title if any usable
+                var modified = false
+                if draftInContext.subject.isEmpty,
+                   let title = textAttachment.title {
+                    draftInContext.subject = title
+                    modified = true
+                }
+
+                // Url if any after a minimalistic input sanitising
+                if let body = textAttachment.body,
+                   let bodyUrl = URL(string: body),
+                   !body.isEmpty {
+                    draftInContext.body = "<a href=\"\(bodyUrl.absoluteString)\">"
```

Comment:
@valentinperignon This works quite well to paste an URL in a draft, but would you recomment something else ?

Reply:
Nice! You can wrap the a in a div, then remove the br
The editor might be happy like this